pennylane_lightning/src/util/Kokkos_Sparse.hpp

```diff
@@ -131,15 +131,18 @@ void apply_Sparse_Matrix_Kokkos(
         const_data_view_type<fp_precision> vector_view(
             reinterpret_cast<const Kokkos::complex<fp_precision> *>(vector_ptr),
             vector_size);
-        Kokkos::View<data_type<fp_precision> *> result_view("rhs", vector_size);
+        std::vector<Kokkos::complex<fp_precision>> result_view_vector(
+            vector_size);
+        data_view_type<fp_precision> result_view(result_view_vector.data(),
+                                                 vector_size);
```

Comment:
We need to allocate memory (in this case with a std::vector), to initialize an (unmanaged) view, for Kokkos Kernels KokkosSparse::spmv method.

Reply:
So this is a work-space then for the operation, rather than a copied block for the data to be added to before the computation?

Reply:
This is where the final result will be copied, by Kokkos Sparse. 
In our context though, this is a workspace, because we'll typecast it to std vector again to output it.

Reply:
Kind of a workspace.

Reply:
Is it possible to overwrite any of the incoming memory blocks with this result, to avoid the allocation, or is it just easiest to do this and move it back when done?